lib/server.rb

```diff
@@ -39,6 +39,67 @@ def scope_help(scope)
       Kramdown::Document.new(text).to_html
     end
 
+    def safe_length(object, attribute)
+      if object && collection = object.send(attribute)
+        collection.length
+      else
+        0
+      end
+    end
+
+    def only_in_a
+      "<h3>Only in '#{@description_a.name}':</h3>"
+    end
+
+    def only_in_b
+      "<h3>Only in '#{@description_b.name}':</h3>"
+    end
+
+    def in_both
+      "<h3>In both descriptions:</h3>"
+    end
+
+    def changed
+      "<h3>In both with different attributes:</h3>"
+    end
+
+    def changed_packages
+      changed = []
+      @diff["packages"].changed.each do |change|
+        changes = []
+        relevant_attributes = ["version", "vendor", "arch"]
+
+        if change[0].version == change[1].version
+          relevant_attributes.push("release")
+          if change[0].release == change[1].release
+            relevant_attributes.push("checksum")
+          end
+        end
+
+        relevant_attributes.each do |attribute|
+          if change[0][attribute] != change[1][attribute]
+            changes.push(attribute + ": " + change[0][attribute] + " ↔ " + change[1][attribute])
+          end
+        end
+
+        changed.push(change[0].name + " (" + changes.join(", ") + ")")
+      end
+      changed
+    end
+
+    def diffable_unmanaged_files
```

Comment:
Assignment Branch Condition size for diffable_unmanaged_files is too high. [19.65/15]